### PR TITLE
work around hang starting database on windows

### PR DESCRIPTION
**Pull request by Dridus**
on windows, it appears that for whatever reason `pg_ctl start` never closes the `stdout`/`stderr` pipes, resulting in a timeout if that's configured or hang if not. I'm not sure the precise reason for this, I'm guessing because `pg_ctl` forks the server and accidentally hands it `stdout`/`stderr` or something along those lines.

I was tipped off to this behavior by `pg-embed`: https://github.com/faokunega/pg-embed/blob/master/src/command_executor.rs#L180-L192

`pg-embed` just doesn't read the output of subprocesses it executes, but this library appears to (e.g. for `database_exists`) and so I hacked around it by abandoning the pipe reads 50ms after the subprocess produces an exit status.

this is clearly a horrible hack, and as written it also adds 50ms to every process invocation on windows, not just `pg_ctl start`. I'm submitting it in the hopes that it's better than broken or at least as a very detailed issue submission.

thanks for the library! sorry for the gross hack

### CODE REVIEW

**Comment by brianheineman** [comment]
Hello @Dridus, thank you for the PR!  I noticed that there is also a suggested style change in this PR to remove usage of the custom `Result`.  Would you mind removing that change from this PR as I believe that is a separate issue from the Windows bug.  I'd be be happy to discuss a change in the interfaces to remove the custom `Result` in a separate issue/PR for all the API's if you would like; I'd be interested to hear the pro's/con's of the change from your perspective.

**Comment by Dridus** [comment]
so (un)revised.

my rationale for it was because I was referring to the `std::result::` version several times (more in some WIP versions) and it got quite noisy, and I prefer to not pun prelude imported things to minimize confusion. I probably won't put in an issue or PR for that at the moment, but I'll keep it in mind for future if I'm making future suggestions!

**Comment by Dridus** [comment]
also thanks for the quick PR responses!

**Review by brianheineman — APPROVED** [review]

File: postgresql_commands/src/traits.rs [hunk 1]
```diff
@@ -160,28 +160,103 @@ impl CommandExecutor for std::process::Command {
...
+            let Some(reader) = reader.as_mut() else {
+                return Ok(String::new());
+            };
+            let mut vec = Vec::new();
+            loop {
+                use tokio::io::AsyncReadExt as _;
+                tokio::select! {
+                    n = reader.read_buf(&mut vec) => {
+                        if n? == 0 {
+                            return Ok(String::from_utf8_lossy(&*vec).into_owned());
```
Comment (brianheineman) [inline postgresql_commands/src/traits.rs:177]:
```
error: deref which would be done by auto-deref
   --> postgresql_commands/src/traits.rs:177:63
    |
[17](https://github.com/theseus-rs/postgresql-embedded/actions/runs/9178579262/job/25238798922?pr=67#step:5:18)7 | ...                   return Ok(String::from_utf8_lossy(&*vec).into_owned());
    |                                                         ^^^^^ help: try: `&vec`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#explicit_auto_deref
    = note: `-D clippy::explicit-auto-deref` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::explicit_auto_deref)]`
```

```suggestion
                            return Ok(String::from_utf8_lossy(&vec).into_owned());
```

File: postgresql_commands/src/traits.rs [hunk 2]
```diff
@@ -160,28 +160,103 @@ impl CommandExecutor for std::process::Command {
...
+                        }
+                    },
+                    _ = exit_anyway_broadcast_receiver.recv() => {
+                        return Ok(String::from_utf8_lossy(&*vec).into_owned());
```
Comment (brianheineman) [inline postgresql_commands/src/traits.rs:181]:
```
error: deref which would be done by auto-deref
   --> postgresql_commands/src/traits.rs:181:59
    |
181 |                         return Ok(String::from_utf8_lossy(&*vec).into_owned());
    |                                                           ^^^^^ help: try: `&vec`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#explicit_auto_deref
```

```suggestion
                        return Ok(String::from_utf8_lossy(&vec).into_owned());
```

**Review by brianheineman — COMMENTED** [review]
There are a couple of linting errors that need to be addressed before merge.

**Comment by brianheineman** [comment]
This change was released in version [0.9.3](https://github.com/theseus-rs/postgresql-embedded/releases/tag/v0.9.3); thank you for your contribution!

**Pull request merged**